app/script/event/EventRepository.js

```diff
@@ -647,79 +647,86 @@ z.event.EventRepository = class EventRepository {
    * @returns {Promise} Resolves with the saved event
    */
   _handleEventSaving(event, source) {
-    const {conversation: conversationId, data: mappedData} = event;
-    const eventId = (mappedData && mappedData.replacing_message_id) || event.id;
-
-    const loadPromise = eventId
-      ? this.conversationService.load_event_from_db(conversationId, eventId)
-      : Promise.resolve();
-
-    return loadPromise.then(storedEvent => {
-      if (!storedEvent) {
-        return this.conversationService.save_event(event);
+    const conversationId = event.conversation;
+    const mappedData = event.data || {};
+
+    //first check if a message that should be replaced exists in DB
+    const findEventToReplacePromise = mappedData.replacing_message_id
+      ? this.conversationService.load_event_from_db(mappedData.replacing_message_id)
+      : Promise.resolve(undefined);
+
+    return findEventToReplacePromise.then(eventToReplace => {
+      const hasLinkPreview = mappedData.previews && mappedData.previews.length;
+      if (!eventToReplace && mappedData.replacing_message_id && !hasLinkPreview) {
```

Comment:
I suggest to also create a boolean variable for `!eventToReplace && mappedData.replacing_message_id`.